### PR TITLE
feat: redesign metrics storage bucket configuration and add resources variables

### PR DESCRIPTION
**Pull request by lentidas**
This PR removes the bucket generation from inside the module. I chose to still provide multiple variants depending on the provider simply because I find it easier to pass the bucket configuration. What I mean is that instead of passing the block in the same way that Thanos would require, I find it best that we only pass the settings required and the module contains the static configuration needed for each variant.

### CODE REVIEW

**Comment by lentidas** [comment]
Still need to update the documentation for the changes that were made.

File: aks/extra-variables.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+variable "metrics_storage" {
+  description = "Azure Blob Storage configuration values for the storage container where the archived metrics will be stored."
```
Comment (fsismondi) [inline aks/extra-variables.tf:2]: shouldnt we add `sensitive` here?

**Review by modridi — DISMISSED** [review]

**Comment by modridi** [comment]
This works on Azure, thanks!

File: aks/extra-variables.tf (hunk 1 above)
Comment (lentidas) [reply aks/extra-variables.tf:2]: We will need to better discuss our approach to sensitive secrets, as per our thread in our Slack channel.

**Review by modridi — APPROVED** [review]

**Pull request merged**